packages/camera/tizen/src/camera_plugin.cc

```diff
@@ -0,0 +1,159 @@
+#include "log.h"
+
+#define CAMERA_CHANNEL_NAME "plugins.flutter.io/camera"
+#define IMAGE_STREAM_CHANNEL_NAME "plugins.flutter.io/camera/imageStream"
+
+class CameraPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(
+      flutter::PluginRegistrar *registrar,
+      FlutterTextureRegistrar *texture_registrar) {
+    auto camera_channel =
+        std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+            registrar->messenger(), CAMERA_CHANNEL_NAME,
+            &flutter::StandardMethodCodec::GetInstance());
+
+    auto camera_plugin =
+        std::make_unique<CameraPlugin>(registrar, texture_registrar);
+
+    camera_channel->SetMethodCallHandler(
+        [plugin_pointer = camera_plugin.get()](const auto &call, auto result) {
+          plugin_pointer->HandleMethodCall(call, std::move(result));
+        });
+
+    registrar->AddPlugin(std::move(camera_plugin));
+  }
+
+  CameraPlugin(flutter::PluginRegistrar *registrar,
+               FlutterTextureRegistrar *texture_registrar)
+      : registrar_(registrar), texture_registrar_(texture_registrar) {}
+
+  virtual ~CameraPlugin() {}
+
+ private:
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    std::string method_name = method_call.method_name();
+
+    LOG_DEBUG("method_name[%s]", method_name.data());
+    if (method_name == "availableCameras") {
+      flutter::EncodableValue availableCameras =
+          CameraDevice::GetAvailableCameras();
+      result->Success(availableCameras);
+    } else if (method_name == "initialize") {
+      // TODO : Request permissions
+      if (method_call.arguments()) {
+        flutter::EncodableMap arguments =
+            std::get<flutter::EncodableMap>(*method_call.arguments());
+
+        auto camera_name_value =
+            arguments[flutter::EncodableValue("cameraName")];
+        std::string camera_name;
+        if (!camera_name_value.IsNull()) {
+          camera_name = std::get<std::string>(camera_name_value);
+        }
+
+        auto resolution_preset_value =
+            arguments[flutter::EncodableValue("resolutionPreset")];
+        std::string resolution_preset;
+        if (!resolution_preset_value.IsNull()) {
+          resolution_preset = std::get<std::string>(resolution_preset_value);
+        }
+
+        auto enable_audio_value =
+            arguments[flutter::EncodableValue("enableAudio")];
+        bool enable_audio;
+        if (!resolution_preset_value.IsNull()) {
+          enable_audio = std::get<bool>(enable_audio_value);
+        }
+
+        LOG_DEBUG("camera_name[%s], resolution_preset[%s], enableAudio[%d]",
+                  camera_name.data(), resolution_preset.data(), enable_audio);
+        flutter::EncodableValue reply = InitializeCameraDevice(
+            camera_name, resolution_preset, enable_audio);
+        result->Success(reply);
```

Comment:
isn't camera initialization asynchronous in Tizen? I mean, user should wait for a state change. If that's true, couldn't we postpone reporting success until the camera is really ready?

Reply:
This plugin works with asynchronous communication over channels. so I think that it doesn't matter whether Tizen camera APIs are asynchronous or not. I think you don't have to seriously consider the future work at this stage since concrete implementations will be added over and over again. thank you for review